src/prefect/workers/base.py

```diff
@@ -1064,6 +1069,14 @@ def _emit_flow_run_executed_event(
             follows=submitted_event,
         )
 
+    def _emit_worker_heartbeat_event(self, heartbeat_origin: str) -> Event:
+        return emit_event(
+            "prefect.worker.heartbeat",
```

Comment:
In addition to create two different event types, could we name these events something like `prefect.worker.poll.flow-run` and `prefect.worker.poll.cancellation`? Workers already have a concept of a heartbeat where they call the `/workers/heartbeat` of a work pool, and these events seem related to a different piece of worker functionality. 

It also might be worthwhile to emit a `prefect.worker.heartbeat` event whenever a worker calls the `/workers/heartbeat` endpoint.

Reply:
ah yes, this is what i feared. ok ill disambiguate

Reply:
> It also might be worthwhile to emit a prefect.worker.heartbeat event whenever a worker calls the /workers/heartbeat endpoint.

What's the heartbeat for? Should we emit this one server-side?

Reply:
@madkinsz i dont think we had a use case for the heartbeat event, probably more anticipatory. i do like the idea of dispatching it at the server endpoint for the existing heartbeat call, WDYT @desertaxle ?

Reply:
Yeah, I agree that it makes more sense to emit it server-side. The heartbeat is so that we know which workers are polling a work pool and the last time they polled. With the introduction of these events we may not need it anymore for Cloud, but we'll want to keep the current heartbeat implementation for OSS.